src/Microsoft.ML.CpuMath/SseIntrinsics.cs

```diff
@@ -0,0 +1,536 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.X86;
+
+namespace Microsoft.ML.Runtime.Internal.CpuMath
+{
+    internal static class SseIntrinsics
+    {
+        private static unsafe Vector128<float> Load1(float* src, int* idx)
+        {
+            return Sse.SetScalarVector128(src[idx[0]]);
+        }
+
+        private static unsafe Vector128<float> Load4(float* src, int* idx)
+        {
+            return Sse.SetVector128(src[idx[3]], src[idx[2]], src[idx[1]], src[idx[0]]);
+        }
+
+        private static Vector128<float> Rotate(Vector128<float> x)
+        {
+            return Sse.Shuffle(x, x, 0x39);
+        }
+
+        private static Vector128<float> RotateReverse(Vector128<float> x)
+        {
+            return Sse.Shuffle(x, x, 0x93);
+        }
+
+        // Warning: this operation changes the value of x => do not reuse x
+        private static unsafe void Store4(Vector128<float> x, float* dst, int* idx)
+        {
+            Sse.StoreScalar(dst + idx[0], x);
```

Comment:
This method should just be a single `Sse.Store`, which will store all four elements at once.

Reply:
Nevermind, just realized that this is for storing four non-linear indices.